kura/org.eclipse.kura.driver.gpio.provider/src/main/java/org/eclipse/kura/internal/driver/gpio/GPIODriver.java

```diff
@@ -245,8 +244,10 @@ private synchronized void runWriteRequest(GPIORequestInfo requestInfo) {
                 record.setChannelStatus(new ChannelStatus(ChannelFlag.FAILURE, message.readFailed(), null));
                 record.setTimestamp(System.currentTimeMillis());
                 logger.warn(message.writeFailed());
-                return;
             }
+        } else {
+            record.setChannelStatus(new ChannelStatus(ChannelFlag.FAILURE, message.readFailed(), null));
```

Comment:
Can this code be externalised to a method to prevent code duplication?

Reply:
👍